crates/bevy_hierarchy/src/one_to_many/one.rs

```diff
@@ -0,0 +1,200 @@
+#[cfg(feature = "reflect")]
+use bevy_ecs::reflect::{
+    ReflectComponent, ReflectFromWorld, ReflectMapEntities, ReflectVisitEntities,
+    ReflectVisitEntitiesMut,
+};
+use bevy_ecs::{
+    component::{Component, ComponentId},
+    entity::{Entity, VisitEntities, VisitEntitiesMut},
+    event::Events,
+    traversal::Traversal,
+    world::{DeferredWorld, FromWorld, World},
+};
+use core::{fmt::Debug, marker::PhantomData, ops::Deref};
+
+use super::{OneToMany, OneToManyEvent};
+
+/// Represents one half of a one-to-many relationship between an [`Entity`] and some number of other [entities](Entity).
+///
+/// The type of relationship is denoted by the parameter `R`.
+#[derive(Component)]
+#[component(
+    on_insert = Self::associate,
+    on_replace = Self::disassociate,
+)]
+#[cfg_attr(feature = "reflect", derive(bevy_reflect::Reflect))]
+#[cfg_attr(
+    feature = "reflect",
+    reflect(
+        Component,
+        MapEntities,
+        VisitEntities,
+        VisitEntitiesMut,
+        PartialEq,
+        Debug,
+        FromWorld
+    )
+)]
+pub struct ManyToOne<R> {
+    entity: Entity,
+    #[cfg_attr(feature = "reflect", reflect(ignore))]
+    _phantom: PhantomData<fn(&R)>,
+}
+
+impl<R: 'static> ManyToOne<R> {
+    fn associate(mut world: DeferredWorld<'_>, a_id: Entity, _component: ComponentId) {
+        world.commands().queue(move |world: &mut World| {
+            let b = world
```

Comment:
I do agree, I struggled with this as I was writing it. After some deliberation, I've decided on `primary` and `foreign` instead of `a` and `b`, and I've included some documentation explaining what those names mean, and their lineage (relational database terminology)